ui/ui-app/src/utils/rest.utils.ts

```diff
@@ -3,9 +3,63 @@ import { ContentTypes } from "@models/contentTypes.model.ts";
 import { AuthService } from "@apicurio/common-ui-components";
 import { Buffer } from "buffer";
 
-const AXIOS = axios.create();
+import { AnonymousAuthenticationProvider, AuthenticationProvider, Headers, RequestInformation } from "@microsoft/kiota-abstractions";
+import { ConfigService } from "@services/useConfigService";
+import { ApicurioRegistryClient, createApicurioRegistryClient } from "@apicurio/apicurio-registry-client/src-generated/registry-client/apicurioRegistryClient";
+import { FetchRequestAdapter } from "@microsoft/kiota-http-fetchlibrary";
+
+// START - Kiota Machinery
+export class TokenAuthenticationProvider implements AuthenticationProvider {
+    private readonly key: string;
+    private readonly accessTokenProvider: () => Promise<string>;
+    private static readonly authorizationHeaderKey = "Authorization";
+    public constructor(key: string, accessTokenProvider: () => Promise<string>) {
+        this.key = key;
+        this.accessTokenProvider = accessTokenProvider;
+    }
+
+    public authenticateRequest = async (request: RequestInformation, additionalAuthenticationContext?: Record<string, unknown>): Promise<void> => {
+        if (!request) {
+            throw new Error("request info cannot be null");
+        }
+        if (additionalAuthenticationContext?.claims && request.headers.has(TokenAuthenticationProvider.authorizationHeaderKey)) {
+            request.headers.delete(TokenAuthenticationProvider.authorizationHeaderKey);
+        }
+        if (!request.headers || !request.headers.has(TokenAuthenticationProvider.authorizationHeaderKey)) {
+            const token = await this.accessTokenProvider();
+            if (!request.headers) {
+                request.headers = new Headers();
+            }
+            if (token) {
+                request.headers.add(TokenAuthenticationProvider.authorizationHeaderKey, `${this.key} ${token}`);
+            }
+        }
+    };
+}
+
+export function createAuthProvider(auth: AuthService): AuthenticationProvider {
+    if (auth.isOidcAuthEnabled()) {
+        return new TokenAuthenticationProvider("Bearer", async () => auth.getToken().then(v => v!));
```

Comment:
Just to be sure, are we testing those configurations of the UI in CI?

Reply:
What configurations?  We're only doing basic testing of the UI in CI - in two places.  We have UI tests in the registry repo, and we have some in the 3scale dev environment as well.

More UI testing is for sure needed - currents tests are mostly smoke tests.

Reply:
Do we test the UI in an environment where OIDC is enables and Keycloak configured?

Reply:
In the 3scale dev environment, yes.  But not in registry CI.